src/modules/producer.js

```diff
@@ -118,7 +119,7 @@ function checkRpc (queue, msg, options) {
 
         if (publishOrSendToQueue.call(this, queue, msg, options)) {
             //defered promise that will resolve when response is received
-            let responsePromise = Promise.defer();
+            let responsePromise =  new Deferred();
```

Comment:
@PuKoren If you want to go ahead for a release with this to be compatible with Node v7 then i will approve and I will create a task for refactor.